### PR TITLE
BUG: BusinessDay addition with non-nano

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -1849,7 +1849,7 @@ cdef class BusinessDay(BusinessMixin):
         reso = get_unit_from_dtype(dtarr.dtype)
         res = self._shift_bdays(i8other, reso=reso)
         if self.offset:
-            res = res.view(dtarr.dtype) + Timedelta(self.offset)
+            res = res.view(dtarr.dtype) + Timedelta(self.offset).as_unit("s")
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/offsets.pyx:1852]: i might be missing something, but wouldn't this regress if the offset contains sub seconds?

```python
idx = pd.date_range("2010/02/01", "2010/02/10", freq="12h", unit='ns')
off = BDay(offset=pd.Timedelta(3, unit="ns"))
t1 = idx + off
```

Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx:1852]: dangit good catch, will update

Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx]: updated+greenish

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**